### PR TITLE
feat(launch): connect to process.env.SELENIUM_REMOTE_URL instead of local launch

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: src/server/chromium/chromium.ts [hunk 1]
```diff
@@ -128,7 +131,77 @@ export class Chromium extends BrowserType {
...
+    const value = JSON.parse(response).value;
+    const sessionId = value.sessionId;
+    progress.log(`<connected to selenium> sessionId=${sessionId}`);
+
+    const disconnectFromSelenium = async () => {
+      progress.log(`<disconnecting from selenium> sessionId=${sessionId}`);
+      await fetchData({
+        url: hubUrl + 'session/' + sessionId,
+        method: 'DELETE',
+      }).catch(() => {});
```
Comment (yury-s) [inline src/server/chromium/chromium.ts:170]: Let's print some progress message on error.

File: src/server/chromium/chromium.ts [hunk 2]
```diff
@@ -128,7 +131,77 @@ export class Chromium extends BrowserType {
...
+
+    try {
+      const capabilities = value.capabilities;
+      const maybeChromeOptions = capabilities['goog:chromeOptions'];
+      const chromeOptions = maybeChromeOptions && typeof maybeChromeOptions === 'object' ? maybeChromeOptions : undefined;
+      const debuggerAddress = chromeOptions && typeof chromeOptions.debuggerAddress === 'string' ? chromeOptions.debuggerAddress : undefined;
+      const chromeOptionsURL = typeof maybeChromeOptions === 'string' ? maybeChromeOptions : undefined;
+      let endpointURL = capabilities['se:cdp'] || debuggerAddress || chromeOptionsURL;
+      if (!['ws://', 'wss://', 'http://', 'https://'].some(protocol => endpointURL.startsWith(protocol)))
+        endpointURL = 'http://' + endpointURL;
```
Comment (yury-s) [inline src/server/chromium/chromium.ts:182]: log this `endpointURL` too?

File: tests/config/setup.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const downloadDir = path.join(assets, 'selenium-grid');
+export const standalone314159 = path.join(downloadDir, 'standalone-3.141.59.jar');
+export const selenium400rc1 = path.join(downloadDir, 'selenium-4.0.0-rc-1.jar');
+export const chromeDriver = path.join(downloadDir, process.platform === 'win32' ? 'chromedriver.exe' : 'chromedriver');
+const chromeDriverZip = path.join(downloadDir, 'chromedriver.zip');
+
+export default async () => {
+  await fs.promises.mkdir(assets, { recursive: true });
+  if (!fs.existsSync(standalone314159))
+    await downloadFile('https://github.com/SeleniumHQ/selenium/releases/download/selenium-3.141.59/selenium-server-standalone-3.141.59.jar', standalone314159);
```
Comment (yury-s) [inline tests/config/setup.ts:38]: Why not save to assets to avoid repeating downloads?

File: tests/config/setup.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  if (!fs.existsSync(selenium400rc1))
+    await downloadFile('https://github.com/SeleniumHQ/selenium/releases/download/selenium-4.0.0-rc-1/selenium-server-4.0.0-rc-1.jar', selenium400rc1);
+  if (!fs.existsSync(chromeDriver)) {
+    const chromeDriverPlatform = {
+      'linux': 'linux64',
+      'darwin': 'mac64',
+      'win32': 'win32',
+    }[process.platform];
+    await downloadFile(`https://chromedriver.storage.googleapis.com/93.0.4577.63/chromedriver_${chromeDriverPlatform}.zip`, chromeDriverZip);
```
Comment (yury-s) [inline tests/config/setup.ts:47]: Shouldn't this url depend on the current Chromium version?

File: tests/browsertype-launch-selenium.spec.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { playwrightTest as test, expect } from './config/browserTest';
+import { standalone314159, selenium400rc1, chromeDriver } from './config/setup';
```
Comment (yury-s) [inline tests/browsertype-launch-selenium.spec.ts:18]: nit: standalone_3_141_59 ?

**Review by yury-s — APPROVED** [review]

File: src/server/chromium/chromium.ts (hunk 2 above)
Comment (dgozman) [reply src/server/chromium/chromium.ts:182]: `connectOverCDP` does it for us already.

File: utils/roll_browser.js [hunk 6]
```diff
@@ -87,6 +87,15 @@ Example:
       process.stdout.write(execSync('npm run --silent doc'));
     } catch (e) {
     }
+
+    if (browserName === 'chromium') {
+      // 5. Update chromedriver.
```
Comment (yury-s) [inline utils/roll_browser.js:92]: omg

**Review by yury-s — APPROVED** [review]

**Pull request merged**